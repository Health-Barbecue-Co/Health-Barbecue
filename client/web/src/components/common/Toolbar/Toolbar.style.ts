import { Theme, createStyles } from '@material-ui/core'

export default (theme: Theme) =>
  createStyles({
    toolbar: {
      background: theme.palette.secondary.main,
      color: theme.palette.secondary.contrastText,
    },
    title: {
      flexGrow: 1,
    },
  })
