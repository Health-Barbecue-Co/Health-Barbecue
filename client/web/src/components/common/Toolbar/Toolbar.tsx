import React, { ReactNode, Fragment } from 'react'
import {
  makeStyles,
  Toolbar as ToolbarCore,
  Typography,
} from '@material-ui/core'

import style from './Toolbar.style'

const useStyle = makeStyles(style)

type ToolbarProps = {
  label: string
  rightActions?: ReactNode[]
  leftActions?: ReactNode[]
}

export const Toolbar: React.FC<ToolbarProps> = (props: ToolbarProps) => {
  const { label, rightActions, leftActions } = props
  const classes = useStyle()

  return (
    <ToolbarCore className={classes.toolbar}>
      {leftActions
        ? leftActions.map((elt, idx) => (
            <Fragment key={`toolbar-leftAction-${idx + 1}`}>{elt}</Fragment>
          ))
        : null}
      <Typography variant="h6" className={classes.title}>
        {label}
      </Typography>
      {rightActions
        ? rightActions.map((elt, idx) => (
            <Fragment key={`toolbar-rightAction-${idx + 1}`}>{elt}</Fragment>
          ))
        : null}
    </ToolbarCore>
  )
}
