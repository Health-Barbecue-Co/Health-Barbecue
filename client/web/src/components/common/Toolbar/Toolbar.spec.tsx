import React from 'react'
import renderer from 'react-test-renderer'

import { Toolbar } from './Toolbar'

describe('Toolbar', () => {
  it('contains a label which is displayed', () => {
    const wrapper = renderer.create(<Toolbar label="hello" />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('contains a leftAction array as props', () => {
    const wrapper = renderer.create(
      <Toolbar label="hello" leftActions={[<span>left</span>]} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('contains a rightAction array as props', () => {
    const wrapper = renderer.create(
      <Toolbar label="hello" rightActions={[<span>right</span>]} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
