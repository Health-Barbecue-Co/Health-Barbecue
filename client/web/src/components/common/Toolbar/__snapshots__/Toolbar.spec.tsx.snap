// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar contains a label which is displayed 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-1 MuiToolbar-gutters"
>
  <h6
    className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
  >
    hello
  </h6>
</div>
`;

exports[`Toolbar contains a leftAction array as props 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-1 MuiToolbar-gutters"
>
  <span>
    left
  </span>
  <h6
    className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
  >
    hello
  </h6>
</div>
`;

exports[`Toolbar contains a rightAction array as props 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-1 MuiToolbar-gutters"
>
  <h6
    className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
  >
    hello
  </h6>
  <span>
    right
  </span>
</div>
`;
